pkg/services/baremetal/host/host.go

```diff
@@ -588,6 +588,31 @@ func (s *Service) actionRegistering() actionResult {
 		return s.recordActionFailure(infrav1.RegistrationError, err.Error())
 	}
 
+	// Check RAID for the second time.
+	// This happens for new machines which don't have rootDeviceHints.
```

Comment:
this comment is wrong. If you reach this check, then rootDeviceHints are actually set. They were not set at the time when the host was chosen to provision it.

Reply:
I removed that line:

```
This happens for new machines which don't have rootDeviceHints.
```